src/common/mailingListServices.jsx

```diff
@@ -0,0 +1,145 @@
+import React from "react"
+import { useState } from "react"
+import Alert from "./Alert"
+
+/**
+ * Test that an email address is in a valid format
+ * @param {string} email
+ * @returns boolean
+ */
+function validateEmailFormat(email) {
+  const pattern =
+    /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+  return pattern.test(String(email).toLowerCase())
+}
+
+/**
+ * Component used to display feedback about the outcome of a subscription request
+ * @param {string} invalid 
+ * @param {string} error 
+ * @param {string} success 
+ */
+const StatusMessage = ({ invalid, error, success }) => {
+  /**
+   * Wrapper for displaying status alert
+   * @param {string} type Determines the Alert color scheme
+   * @param {string} message Text content
+   */
+  const Status = ({ type, message }) => {
+    if (!message) return
+    return (
+      <Alert type={type}>
+        <>{message}</>
```

Comment:
don't think we need fragments here do we? since it's wrapped inside `Alert` as parent already.